packages/@ourworldindata/components/src/markdown/remarkPlainLinks.ts

```diff
@@ -0,0 +1,23 @@
+import findAndReplace from "mdast-util-find-and-replace"
+
+export const urlRegex = /https?:\/\/([\w-]+\.)+[\w-]+(\/[\w\- .\+/?:%&=~#]*)?/
```

Comment:
Ah, now that I think of it, including `.?:` can be annoying in cases like `Learn more at https://ourworldindata.com/poverty.`.

I know that some other sites do something like "match `.?:` unless they are at the very end of the URL", which seems sensible.

Reply:
Good catch. I took this opportunity to also make sure we match urls with unicode word characters in the path. 

AFAIK to properly match this with a regex you need to use negative lookbehind assertions and they are not supported in Safari before 16.4 which came out in Spring this year so I instead handle trailing punctuation characters in the replacement function.